src/system/MailerModule/Form/Type/ConfigType.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * This file is part of the Zikula package.
+ *
+ * Copyright Zikula Foundation - http://zikula.org/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zikula\MailerModule\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Zikula\Bundle\CoreBundle\DynamicConfigDumper;
+use Zikula\Common\Translator\TranslatorInterface;
+use Zikula\Common\Translator\TranslatorTrait;
+use Zikula\ExtensionsModule\Api\VariableApi;
+
+/**
+ * Configuration form type class.
+ */
+class ConfigType extends AbstractType
+{
+    use TranslatorTrait;
+
+    /**
+     * @var array
+     */
+    protected $dataValues;
+
+    /**
+     * ConfigType constructor.
+     *
+     * @param TranslatorInterface $translator   Translator service instance.
+     * @param DynamicConfigDumper $configDumper DynamicConfigDumper service instance.
+     * @param VariableApi         $variableApi  VariableApi service instance.
+     */
+    public function __construct(TranslatorInterface $translator, DynamicConfigDumper $configDumper, VariableApi $variableApi)
+    {
+        $this->setTranslator($translator);
+
+        $params = $configDumper->getConfiguration('swiftmailer');
+        $modVars = $variableApi->getAll('ZikulaMailerModule');
+        $this->dataValues = array_merge($params, $modVars);
+    }
+
+    /**
+     * Sets the translator.
+     *
+     * @param TranslatorInterface $translator Translator service instance.
+     */
+    public function setTranslator(/*TranslatorInterface */$translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('transport', 'Symfony\Component\Form\Extension\Core\Type\ChoiceType', [
+                'label' => $this->__('Mail transport'),
+                'data' => $this->dataValues['transport'],
+                'empty_data' => 'mail',
+                'choices' => [
+                    $this->__('Internal PHP `mail()` function') => 'mail',
+                    $this->__('Sendmail message transfer agent') => 'sendmail',
+                    $this->__('Google gmail') => 'gmail',
+                    $this->__('SMTP mail transfer protocol') => 'smtp',
+                    $this->__('Development/debug mode (Do not send any email)') => 'test'/*'null'*/
+                ],
+                'choices_as_values' => true
+            ])
+            ->add('charset', 'Symfony\Component\Form\Extension\Core\Type\TextType', [
+                'label' => $this->__('Character set'),
+                'data' => $this->dataValues['charset'],
```

Comment:
setting `'data'` in the form doesn't do what you "hope" - e.g. - it does NOT set default data 😢 I have found there is no "good" way to set default values (I've researched it a lot actually). Instead, simply set the module vars as the values when setting up the forum

    $form = $this->createForm('Zikula\MailerModule\Form\Type\ConfigType', $this->getVars());

and remove the `'data'` option from the form.